raiden/network/transport/matrix/transport.py

```diff
@@ -677,6 +687,34 @@ def _initialize_room_inventory(self) -> None:
                 "Found room", room=room, aliases=room.aliases, members=room.get_joined_members()
             )
 
+        self._leave_unexpected_rooms(rooms_to_leave)
+
+    def _extract_partner_addresses(self, members: List[User]) -> List[Address]:
+        assert self._raiden_service is not None, "_raiden_service not set"
+        addresses = [validate_userid_signature(user) for user in members]
+        joined_partner_addresses: List[Address] = list()
+
+        return [
+            address
+            for address in addresses
+            if address not in joined_partner_addresses and self._raiden_service.address != address
+        ]
+
+    def _leave_unexpected_rooms(self, rooms_to_leave: List[Room]) -> None:
+        assert self._raiden_service is not None, "_raiden_service not set"
+
+        for room in rooms_to_leave:
+            log.warning(
```

Comment:
Please use `self.log,` that is a bound logger instance that automatically adds the `node`, `current_user` and `transport_uuid` keys to each log record: https://github.com/raiden-network/raiden/blob/29f94b77eb4963be0d47331c41b5826fbb71cd0e/raiden/network/transport/matrix/transport.py#L390-L394